Source/EventParser.swift

```diff
@@ -36,6 +35,11 @@ class EventParser {
         }
     }
 
+    func reset() {
```

Comment:
We must now be able to explicitly reset this parser when the stream completes. Before we relied on the fact that we were required to dispatch the event a final time when the stream completed which reset the fields. Now we want to just drop any in-progress events when the stream drops.